client/my-sites/site-settings/wrap-settings-form.jsx

```diff
@@ -0,0 +1,169 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { flowRight, omit, memoize } from 'lodash';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { protectForm } from 'lib/protect-form';
+import trackForm from 'lib/track-form';
+import {
+	isRequestingSiteSettings,
+	isSavingSiteSettings,
+	isSiteSettingsSaveSuccessful,
+	getSiteSettings
+} from 'state/site-settings/selectors';
+import { recordGoogleEvent, recordTracksEvent } from 'state/analytics/actions';
+import { saveSiteSettings } from 'state/site-settings/actions';
+import { removeNotice, successNotice, errorNotice } from 'state/notices/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import QuerySiteSettings from 'components/data/query-site-settings';
+
+const wrapSettingsForm = getFormSettings => SettingsForm => {
+	class WrappedSettingsForm extends Component {
+		componentWillMount() {
+			this.props.replaceFields( getFormSettings( this.props.settings ) );
+		}
+
+		componentWillReceiveProps( nextProps ) {
+			if ( nextProps.siteId !== this.props.siteId ) {
+				nextProps.clearDirtyFields();
+			}
+
+			if ( nextProps.settings !== this.props.settings ) {
+				let newState = getFormSettings( nextProps.settings );
+				//If we have any fields that the user has updated,
+				//do not wipe out those fields from the poll update.
+				newState = omit( newState, nextProps.dirtyFields );
+				nextProps.replaceFields( newState );
+			}
+
+			if (
+			this.props.isSavingSettings &&
```

Comment:
The discussion form is not affected by this bug, so we can probably fix this issue as a separate parallel PR.